readers/elasticsearch/src/main/java/com/criteo/hadoop/garmadon/elasticsearch/ElasticSearchReader.java

```diff
@@ -117,7 +117,7 @@ void writeToES(GarmadonMessage msg) {
                 addEventToBulkProcessor(eventMap, msg.getTimestamp(), msg.getCommittableOffset());
             }
         } else {
-            Map<String, Object> eventMap = ProtoConcatenator.concatToMap(Arrays.asList(msg.getHeader(), msg.getBody()), true);
+            Map<String, Object> eventMap = ProtoConcatenator.concatToMap(msg.getTimestamp(), Arrays.asList(msg.getHeader(), msg.getBody()), true);
```

Comment:
please remove  eventMap.put("timestamp", timestamp); in private void addEventToBulkProcessor(Map<String, Object> eventMap, long timestamp, CommittableOffset committableOffset) {

Reply:
And remove msg.getTimestamp in             addEventToBulkProcessor(eventMap, msg.getTimestamp(), msg.getCommittableOffset());

Reply:
Done